test/index.spec.ts

```diff
@@ -0,0 +1,236 @@
+import { expect } from 'chai';
+import type { Interceptable } from 'undici';
+import { MockAgent, setGlobalDispatcher } from 'undici';
+import * as net from 'net';
+import * as fs from 'fs';
+import * as netCheck from '../lib/index';
+
+describe('network-check', () => {
+	const HOST = 'https://balena.io';
+	let mockAgent: MockAgent;
+	let mockPool: Interceptable;
+	before(() => {
+		mockAgent = new MockAgent();
+		setGlobalDispatcher(mockAgent);
+	});
+
+	describe('monitor', () => {
+		beforeEach(() => {
+			mockPool = mockAgent.get(HOST);
+			mockPool
+				.intercept({
+					path: '/test1',
+				})
+				.reply(200)
+				.persist();
+		});
+
+		afterEach(async () => {
+			await mockPool.close();
+		});
+
+		it('throws an error if checkFn is not a function', () => {
+			expect(() => {
+				netCheck.monitor(null as any, { url: HOST }, () => {
+					/* noop */
+				});
+			}).to.throw('checkFn should be a Function');
+		});
+
+		it('calls checkFn with opts at specified interval', async () => {
+			let count = 0;
+			const interval = 100;
+			const opts = { url: `${HOST}/test1`, interval };
+			const checkFn = async () => {
+				count++;
+				return true;
+			};
+
+			netCheck.monitor(checkFn, opts, () => {
+				/* noop */
+			});
+
+			await new Promise((resolve) => setTimeout(resolve, interval * 3));
```

Comment:
Use
```ts
import { setTimeout } from 'timers/promises';
```
and then you can do
```suggestion
			await setTimeout(interval * 3);
```
rather than manually wrapping it